adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -462,23 +467,23 @@ public final void doStop(final Future<Void> stopFuture) {
             log.error("error in preShutdown", e);
         }
 
-        final Future<Void> serverStopTracker = Future.future();
+        final Promise<Void> serverStopTracker = Promise.promise();
         if (server != null) {
             server.close(serverStopTracker);
         } else {
             serverStopTracker.complete();
         }
 
-        final Future<Void> insecureServerStopTracker = Future.future();
+        final Promise<Void> insecureServerStopTracker = Promise.promise();
         if (insecureServer != null) {
             insecureServer.close(insecureServerStopTracker);
         } else {
             insecureServerStopTracker.complete();
         }
 
-        CompositeFuture.all(serverStopTracker, insecureServerStopTracker)
-            .compose(v -> postShutdown())
-            .compose(s -> stopFuture.complete(), stopFuture);
+        CompositeFuture.all(serverStopTracker.future(), insecureServerStopTracker.future())
+        .compose(v -> postShutdown())
```

Comment:
Something with the indentation here looks wrong.

Reply:
hm, this is how Eclipse indents (or rather doesn't indent) it when I hit return at the end of the former line ...

Reply:
That is weird. Try formatting the code, every now and then I see differences between direct editing and formatting afterwards.